cpp/tests/test_shuffler.cpp

```diff
@@ -666,6 +661,9 @@ TEST_P(ShuffleInsertGroupedTest, InsertPackedDataNoHeadroom) {
     // resume progress thread - this will guarantee that shuffler progress function is
     // marked as done. This is important to ensure that the test does not hang.
     progress_thread->resume();
+
+    shuffler.reset();
+    br.reset();
```

Comment:
This must be wrong, otherwise our RAII destructors are broken. What is this for?

Reply:
> This must be wrong, otherwise our RAII destructors are broken. 

@wence- Sorry, could you clarify this comment?

> What is this for?

Previously `br` and `shuffler` were class members, and their destruction happened during the test object destruction. Since we don't create the objects during `Setup`, I thought its best to move the creation and destruction inside the test impl.

Reply:
@madsbk I understand. I was trying to be more explicit. But that's fine, I'll remove these lines.